mirgecom/boundary.py

```diff
@@ -35,15 +41,99 @@
 import numpy as np
 from meshmode.dof_array import thaw
 from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
-from grudge.trace_pair import TracePair
 from mirgecom.fluid import make_conserved
+from grudge.trace_pair import TracePair
+from mirgecom.inviscid import inviscid_facial_flux
 
+from abc import ABCMeta, abstractmethod
 
-class PrescribedBoundary:
-    """Boundary condition prescribes boundary soln with user-specified function.
+
+class FluidBoundary(metaclass=ABCMeta):
+    r"""Abstract interface to fluid boundary treatment.
+
+    .. automethod:: inviscid_boundary_flux
+    """
+
+    @abstractmethod
+    def inviscid_boundary_flux(self, discr, btag, cv, eos, **kwargs):
+        """Get the inviscid flux across the boundary faces."""
+
+
+class FluidBC(FluidBoundary):
+    r"""Abstract interface to boundary conditions.
+
+    .. automethod:: inviscid_boundary_flux
+    .. automethod:: boundary_pair
+    """
+
+    def inviscid_boundary_flux(self, discr, btag, cv, eos, **kwargs):
+        """Get the inviscid part of the physical flux across the boundary *btag*."""
+        raise NotImplementedError()
+
+    def boundary_pair(self, discr, btag, cv, eos, **kwargs):
+        """Get the interior and exterior solution (*u*) on the boundary."""
+        raise NotImplementedError()
```

Comment:
Minor suggestion, but since the BC class is already an abstract class, might as well use the abstract methods? Anyways, consider this an optional request if you don't care for it :P
```suggestion
    @abstractmethod
    def inviscid_boundary_flux(self, discr, btag, cv, eos, **kwargs):
        """Get the inviscid part of the physical flux across the boundary *btag*."""

    @abstractmethod
    def boundary_pair(self, discr, btag, cv, eos, **kwargs):
        """Get the interior and exterior solution (*u*) on the boundary."""
```
?

Reply:
I'm game - bout to push the button.  I fear chaos breaking out.